charts/linkerd2/templates/policy-crd.yaml

```diff
@@ -0,0 +1,290 @@
+                            required: [key, operator, value]
+                            properties:
+                              key:
+                                type: string
+                              operator:
+                                type: string
+                                enum: [In, NotIn]
+                              values:
+                                type: array
+                                items:
+                                  type: string
+
+                client:
+                  description:  Describes clients authorized to access a server.
+                  type: object
+                  oneOf:
+                    - required: [meshTLS]
+                    - required: [unauthenticated]
+                  properties:
+
+                    networks:
+                      description: >-
+                        Limits the client IP addresses to which this
+                        authorization applies. If unset, the server chooses a
+                        default (typically, all IPs or the cluster's pod
+                        network).
+                      type: array
+                      items:
+                        type: object
+                        required: [cidr]
+                        properties:
+                          cidr:
+                            type: string
+                          except:
+                            type: array
+                            items:
+                              type: string
+
+                    unauthenticated:
+                      description: >-
+                        Authorizes unauthenticated clients to access a server.
+                      type: boolean
+
+                    meshTLS:
+                      type: object
+                      oneOf:
+                        - required: [unauthenticatedTLS]
+                        - required: [identities]
+                        - required: [serviceAccounts]
+
+                      properties:
+                        unauthenticatedTLS:
+                          type: boolean
+                          description: >-
+                            Indicates that no client identity is required for
+                            communication.
+
+                            This is mostly important for the identity
+                            controller, which must terminate TLS connections
+                            from clients that do not yet have a certificate.
+
+                        identities:
+                          description: >-
+                            Authorizes clients with the provided proxy identity
+                            strings (as provided via MTLS)
+
+                            The `*` prefix can be used to match all identities in
+                            a domain. An identity string of `*` indicates that
+                            all authentication clients are authorized.
```

Comment:
Wondering why is this handled in separate fields (`identities` and `suffixes`) in `inbound.proto`?

Reply:
The intent is to make the user-facing CRD configuration to be ergonomic: as a user, I just want to list a set of patterns.

But, for the internal proto API, we want the proxy to be crystal clear about what it's being asked to do (exact match vs suffix match). It's the job of the controller to do validation (i.e. ultimately as an admission controller for these resources, though that's not yet been implemented). I definitely do *not* want the proxy to have to deal with ambiguous configuration (parsing strings, etc).